envi/memory.py

```diff
@@ -145,12 +146,15 @@ def probeMemory(self, va, size, perm):
         mmap = self.getMemoryMap(va)
         if mmap is None:
             return False
+
         mapva, mapsize, mapperm, mapfile = mmap
         mapend = mapva+mapsize
         if va+size > mapend:
             return False
-        if mapperm & perm != perm:
+
+        if mapperm & perm != perm and not self._supervisor:
```

Comment:
If we're going to move this here, we need to move the setting of "_supervisor" in the `MemoryObject` constructor into the constructor of IMemory so everything is nice and tidy.

Reply:
you make a great point.